sydent/http/matrixfederationagent.py

```diff
@@ -114,8 +113,7 @@ def __init__(
         #   `None`:      there is no (valid) .well-known here
         self._well_known_cache = _well_known_cache
 
-    @defer.inlineCallbacks
-    def request(
+    async def request(
```

Comment:
This one should actually stay as a `Deferred` -- the `MatrixFederationAgent` object declares itself as implementing the [`IAgent` interface](https://twistedmatrix.com/documents/current/api/twisted.web.iweb.IAgent.html#request), which declares that `request` returns a `Deferred`.

(Interfaces aren't very common in Python, but Twisted uses them, they pretty much describe the API of objects without any implementation. So there can be multiple implementations of `IAgent` that all should act the same.)

In synapse we just sprinkled a few `defer.ensureDeferred` calls into this function. I think that's OK to do here too.

Reply:
Okay makes sense!